server/core/basic_cluster.go

```diff
@@ -24,21 +26,46 @@ import (
 	"go.uber.org/zap"
 )
 
+// SelectConfig provides parameters for selecting regions.
+type SelectConfig struct {
+	NewRegionFirst bool
+	NewProbability float64
+	MaxRegionCount uint64
+}
+
 // BasicCluster provides basic data member and interface for a tikv cluster.
 type BasicCluster struct {
 	sync.RWMutex
-	Stores  *StoresInfo
-	Regions *RegionsInfo
+	Stores     *StoresInfo
+	Regions    *RegionsInfo
+	NewRegions *regionQueue
+	SelectConf *SelectConfig
 }
 
 // NewBasicCluster creates a BasicCluster.
 func NewBasicCluster() *BasicCluster {
 	return &BasicCluster{
-		Stores:  NewStoresInfo(),
-		Regions: NewRegionsInfo(),
+		Stores:     NewStoresInfo(),
+		Regions:    NewRegionsInfo(),
+		NewRegions: newRegionQueue(),
```

Comment:
Can we make new region as an abstract structure?

Reply:
In that case we need two files. A file declares the abstract interface(put it in pd/ pkg/queue?). The other implements the region's queue, just like the regionTree. Would that be better?

Reply:
They can be put under `server/statictics`. You can see how hot region is managed as an example.

Reply:
Hot cache is like new region set. It is of limited size, and keep what schedulers care about.

Reply:
Ok, I'll try.

Reply:
I can't do like this because `import cycle not allowed`. Package `core` has been imported by files in `statistics`, which cannot be imported by files in `core` circularly.